fa-products-service/http-get-single-product/index.ts

```diff
@@ -1,17 +1,63 @@
-import { Product } from '../types';
+import { Product, ProductWithStock, Stock } from '../types';
 import { AzureFunction, Context, HttpRequest } from '@azure/functions';
-import { products } from '../constants';
+import { CosmosClient } from '@azure/cosmos';
+
+const key = process.env.COSMOS_KEY;
+const endpoint = process.env.COSMOS_ENDPOINT;
+
+const databaseName = `products-db`;
+const containerStocksName = `stocks`;
+const containerProductsName = `products`;
+
+const cosmosClient = new CosmosClient({ endpoint, key });
+
+const database = cosmosClient.database(databaseName);
+const containerStocks = database.container(containerStocksName);
+const containerProducts = database.container(containerProductsName);
 
 const getSingleProduct: AzureFunction = async function (
   context: Context,
   req: HttpRequest
 ): Promise<void> {
-  context.log('HTTP trigger getSingleProduct function processed a request.');
-  context.res = {
-    body: products.find(
-      (product) => product.id === `sku${req.params.productId}`
-    ),
-  };
+  if (req.params.productIdOrTotal === 'total') {
+    const responseStocks = await containerStocks.items
+      .query('SELECT * from c')
+      .fetchAll();
+
+    const total = responseStocks.resources.reduce(
+      (previousTotal, singleStock: Stock) => {
+        return singleStock.count + previousTotal;
+      },
+      0
+    );
+    context.res = {
+      body: total,
+    };
+  } else {
+    const productId = req.params.productIdOrTotal;
+    const responseStocks = await containerStocks.items
+      .query(
+        `SELECT s.count FROM stocks s where s.product_id = "sku${productId}"`
+      )
+      .fetchAll();
+
+    const responseProducts = await containerProducts.items
+      .query(
+        `SELECT p.id, p.title, p.price, p.description FROM products p where p.id = "sku${productId}"`
+      )
+      .fetchAll();
+    const singleProduct: ProductWithStock = {

```

Comment:
Nonblocking: you can create a variable like const product = responseProducts.resources[0] as Product to eliminate type several times